src/AquaCalc5000/Parsers/HeaderParser.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Globalization;
+
+namespace AquaCalc5000.Parsers
+{
+    public class HeaderParser
+    {
+        private readonly CsvParser _csvParser;
+
+        public HeaderParser(CsvParser csvParser)
+        {
+            _csvParser = csvParser;
+        }
+
+        public ParsedData Parse()
+        {
+            return new ParsedData
+            {
+                LocationIdentifier = _csvParser.GetRequiredStringByLabel(AquaCalcConstants.GageId),
```

Comment:
We'll need to add a configurable behaviour here. This is my fault. I don't think I spec'd that well enough in the JIRA issue.

The plugin needs a configuration setting, named something like "AssumeUsgsSiteIdentifiers", which should default to true (so Nebraska won't need to change the default behaviour). Other consumers may want to disable the setting.

Nebraska has historically used site identifiers that follow the USGS naming convention, but omit the leading zeros. But their NG system is now using the full 8-digit identifier, for shared sites.

So these AquaCalc files from Nebraska contain `GAGE ID#  ,6690500` but must be mapped to `06690500` (inserting one leading zero) before asking the framework to look up the location.

If the plugin calls `IFieldDataAppender.GetLocationByIdentifier("6690500")`, the framework halts all execution. It does not throw an exception (which the plugin could theoretically catch), but instead the framework just shuts down the appdomain sandbox. The call to GetLocationByIdentifier NEVER RETURNS. It's so gross!

So we'll need some optional padding logic, like this:
```C#
private string ConvertGageIdToLocationIdentifier(string gageId)
{
    if (!Config.AssumeUsgsSiteIdentifiers)
        return gageId;

    return int.TryParse(gageId, out var numericId)
        ? $"{numericId:D8}"
        : gageId;
}
```

The configuration that stores this setting is also fussy. I'd recommend using a JSON config file pattern, like the SxS plugin uses. https://github.com/AquaticInformatics/sxs-pro-field-data-plugin/blob/master/src/SxSPro/Config.cs

This technique will include a Config.json file in the plugin manifest, with the setting defaulting to true,

And someone else can manually tweak the `%ProgramData%\Aquatic Informatics\AQUARIUS Server\FieldDataPlugins\AquaCalc5000\Config.json` file to their needs.